src/nvim/msgpack_rpc/channel.c

```diff
@@ -464,7 +464,11 @@ static void handle_request(Channel *channel, msgpack_object *request)
   }
 
   Array args = ARRAY_DICT_INIT;
-  msgpack_rpc_to_array(msgpack_rpc_args(request), &args);
+  if (!msgpack_rpc_to_array(msgpack_rpc_args(request), &args)) {
+    handler.fn = msgpack_rpc_handle_invalid_arguments;
```

Comment:
Handler for missing method could be replaced by this. Has "invalid arguments" priority over "missing methods"?